app/services/search/search_builder.rb

```diff
@@ -2,62 +2,65 @@ class Search::SearchBuilder
   DEFAULT_HITS_PER_PAGE = 10
   DEFAULT_PAGE = 1
 
-  attr_reader :hits_per_page, :keyword, :location_search, :page, :params_hash,
-              :point_coordinates, :search_filters, :search_replica, :sort_by, :stats, :total_count, :vacancies,
-              :wider_search_suggestions
+  attr_reader :params_hash, :keyword, :page, :hits_per_page, :stats, :total_count, :vacancies
 
   def initialize(form_hash)
     @params_hash = form_hash
     @keyword = @params_hash[:keyword]
     @hits_per_page = (@params_hash[:per_page] || DEFAULT_HITS_PER_PAGE).to_i
     @page = (@params_hash[:page] || DEFAULT_PAGE).to_i
 
-    build_location_search
-    build_search_filters
-    build_search_replica
     call_search
-    build_suggestions if @vacancies.empty? && any?
   end
 
   def only_active_to_hash
-    @params_hash.merge(keyword: @keyword).delete_if do |k, v|
-      v.blank? || (k.eql?(:radius) && @params_hash[:location].blank?) || k.eql?(:jobs_sort) || k.eql?(:page)
+    @params_hash.merge(keyword: @keyword).except(:jobs_sort, :page).reject do |k, v|
```

Comment:
I would prefer:

```
@params_hash.merge(keyword: @keyword)
                          .except(:jobs_sort, :page)
                          .reject { |k, v| v.blank? || (k == :radius && @params_hash[:location].blank?) }
```

Reply:
👍 

To be honest having slept on it I'm also a bit confused as to why this is merging `keyword` back into `@params_hash`... where it came from in the first place!

Reply:
Context, there was some code for a business decision once that updated the keyword when a LocationPolygon was missing. E.g I search for London, but the polygon was missing, so London was added to the keyword. 

This was removed and I assume this slipped through somebody's cracks 😂